app/adapters/abstract_adapter.rb

```diff
@@ -94,9 +94,10 @@ def self.parse_response(response)
 
     content_type = response.content_type.presence or return body
 
-    case Mime::Type.lookup(content_type)
-    when JSON_TYPE then JSON.parse(body)
-    else raise InvalidResponseError.new response: response, message: 'Unknown Content-Type'
+    if Mime::Type.lookup(content_type).match? JSON_TYPE
```

Comment:
```suggestion
    if Mime::Type.lookup(content_type).start_with? JSON_TYPE
```

Reply:
If this code suggestion works, it would be much preferable. Otherwise we should be good with `match?`. Thank you!

Reply:
> If this code suggestion works, it would be much preferable. Otherwise we should be good with `match?`. Thank you!

Well, they are not strings, they are instances of `Mime::Type`, and this `match?` is "special": https://github.com/rails/rails/blob/v7.0.8.1/actionpack/lib/action_dispatch/http/mime_type.rb#L287-L291
It takes into account some "synonyms", which I don't know what it is, and we probably don't need it. Potentially, we could either 1. apply `to_s.downcase` to both and match with `start_with?` as you suggest, or 2. drop he whole `Mime::Type.lookup` thing and work with plain strings.

Reply:
Then keep it as is. Fine with me, thank you!